internal/rbd/controllerserver.go

```diff
@@ -300,6 +306,16 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 		return nil, err
 	}
 
+	// we have to resize the RBD volume with block mode request
+	if expansionRequired {
+		util.DebugLog(ctx, "\n requested volume size: %v is greater than the parent volume, increasing the size", newVolSize)
+		err := rbdVol.resize(newVolSize)
```

Comment:
what if the volume is created successfully and resize fails are we validating the new size and making sure the new requested volume has a new size?